src/app/shared/form/form.component.ts

```diff
@@ -0,0 +1,244 @@
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Params, Router } from '@angular/router';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+
+import { AutoTitleService } from 'src/app/core/_services/shared/autotitle.service';
+import { UnsubscribeService } from 'src/app/core/_services/unsubscribe.service';
+import { AlertService } from 'src/app/core/_services/shared/alert.service';
```

Comment:
Should we switch to snackbar?

Reply:
Change service?

Reply:
Ok I see, you wrap snackBar in a custom service? Yes that's probably a good idea. Never mind then :)